feat(guide-cue): add cues COMPASS-6614

mabaasit opened the pull request:
## Description
<!--- Describe your changes in detail -->
<!--- If applicable, describe (or illustrate) architecture flow -->

### Checklist
- [ ] New tests and/or benchmarks are included
- [ ] Documentation is changed or added
- [ ] I have signed the MongoDB Contributor License Agreement (https://www.mongodb.com/legal/contributor-agreement)

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it's updating a dependancy, link to the Pull Request that originally introduced the fix -->
- [ ] Bugfix
- [x] New feature
- [ ] Dependency update
- [ ] Misc

## Open Questions
<!--- Any particular areas you'd like reviewers to pay attention to? -->

## Dependents
<!--- If applicable, link PRs/commits that this PR is dependent on or is a dependency of. -->

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] *Backport Needed*
- [ ] Patch (non-breaking change which fixes an issue)
- [ ] Minor (non-breaking change which adds functionality)
- [ ] Major (fix or feature that would cause existing functionality to change)

gribnoysup commented on packages/compass-components/package.json:
```diff
@@ -71,6 +71,7 @@
     "@leafygreen-ui/tokens": "^1.4.1",
     "@leafygreen-ui/tooltip": "^9.0.2",
     "@leafygreen-ui/typography": "^15.2.0",
+    "@mongodb-js/compass-logging": "^1.1.6",
```
`compass-logging` creates a dependency on electron and node APIs in a package that we decided should stay unaware of compass internals

mabaasit replied:
that's a good point. having GC tracking outside of this component will make it cumbersome to track when using the component. 
I'll create a follow up ticket as suggested to remove this dependency.

gribnoysup replied:
FWIW this is the way to go about this with the tracking interface we have and we will be doing this for signals as tracking should stay close to the actual tracked use-case instead of being a generic catch-all, otherwise we should've designed the tracking system with a catch-all interface in the first place (it was like that before and it was hard to reason about as this was creating implicit dependencies all over the place)

mabaasit replied:
Removed dependency and added Provider to accomplish this.

gribnoysup commented on packages/compass-components/src/components/guide-cue/guide-cue.tsx:
```diff
@@ -45,8 +82,8 @@ export type GuideCueProps<T> = Omit<
   GroupAndStep & {
-    description: string | React.ReactElement;
     cueId: string;
+    description: string | React.ReactElement;
```
Feel free to ignore, but this (and in a few other places) is basically becoming a ReactChild type

```suggestion
    description: React.ReactChild;
```

gribnoysup commented on packages/compass-components/src/components/item-action-controls.tsx:
```diff
@@ -77,10 +77,29 @@ function actionTestId<Action extends string>(
+}: {
+  glyph?: string | React.ReactElement;
+  size?: ItemActionButtonSize;
+  className?: string;
```
Is this className used for something? I can't see it being passed anywhere

gribnoysup commented on packages/compass-sidebar/src/components/navigation-items.tsx:
```diff
@@ -223,12 +243,12 @@ export function NavigationItems({
 
     return actions;
-  }, [isReadOnly]);
+  }, [isReadOnly, onAction, showCreateDatabaseGuideCue]);
```
Why is `onAction` added to this list?

mabaasit replied:
leftover, removed it!

gribnoysup commented on packages/compass-sidebar/src/components/navigation-items.tsx:
```diff
@@ -258,16 +278,21 @@ export function NavigationItems({
   );
 }
 
+const DEFAULT_SERVER_DATABASES = ['admin', 'config', 'local'];
```
mongodb-ns package provides this metadata (and handles some other cases too that you are not covering here), would it make sense to use it instead of hardcoding this list?

gribnoysup commented on packages/compass-shell/src/components/shell-header/shell-header.jsx:
```diff
@@ -106,7 +105,23 @@ export const ShellHeader = ({
           aria-pressed={isExpanded}
         >
-          &gt;_MONGOSH
+          <GuideCue
```
Now that we have this cue, would it make sense to remove blinking animation from this element?

mabaasit replied:
I didn't notice this cool animation until now!

mabaasit replied:
removed in [8f803c6](https://github.com/mongodb-js/compass/pull/4568/commits/8f803c6c3c54902343319d713a63d3186f51f04c)

gribnoysup commented on packages/compass-home/src/contexts/app-registry-context.ts:
```diff
@@ -1,9 +1,9 @@
 import { createContext, useContext, useState } from 'react';
 import type React from 'react';
 import AppRegistry from 'hadron-app-registry';
-import debugModule from 'debug';
```
You will probably need to remove it from package deps now, doesn't look like it's used anymore in this plugin

gribnoysup commented:
`compass-shell` test failures look legit, the tests are stuck for me locally, have you looked into that?

mabaasit commented on packages/compass-shell/src/components/compass-shell/compass-shell.spec.jsx:
```diff
@@ -20,11 +20,6 @@ const fakeRuntime = {
 describe('CompassShell', function () {
-  before(function () {
-    // https://github.com/jsdom/jsdom/issues/1695
-    HTMLElement.prototype.scrollIntoView = function () {};
```
already added in #4418

gribnoysup reviewed (approved):
Looks good! CI is failing because of the package lock issue, can you sync the branch with main just so we can confirm that all tests are passing now?

mabaasit commented:
I removed this ([efa4c79](https://github.com/mongodb-js/compass/pull/4568/commits/efa4c797cf063f0e55f9a441e993ed594ab46ac4)) fix and added proper clean up and tests for GC in shell-header. I think that's better than disabling GC. Let me know what you think about that. 

And I'll sync the branch

gribnoysup commented:
Nice, yeah, even better 👍
